android/app/src/test/java/org/smartregister/fhircore/auth/account/AccountAuthenticatorTest.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.auth.account
+
+import android.accounts.Account
+import android.accounts.AccountManager
+import android.content.Context
+import android.content.Intent
+import androidx.core.os.bundleOf
+import androidx.test.core.app.ApplicationProvider
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import org.smartregister.fhircore.auth.OAuthResponse
+import org.smartregister.fhircore.robolectric.FhircoreTestRunner
+import org.smartregister.fhircore.shadow.FhirApplicationShadow
+
+@RunWith(FhircoreTestRunner::class)
+@Config(shadows = [FhirApplicationShadow::class])
+class AccountAuthenticatorTest {
+  private lateinit var accountAuthenticator: AccountAuthenticator
+  private lateinit var accountHelper: AccountHelper
+  private lateinit var accountManager: AccountManager
+  private val context = ApplicationProvider.getApplicationContext<Context>()
+
+  lateinit var captor: CapturingSlot<Map<String, String>>
+
+  @Before
+  fun setUp() {
+    accountHelper = mockk()
+    accountManager = mockk()
+
+    accountAuthenticator = AccountAuthenticator(context)
+    accountAuthenticator.accountHelper = accountHelper
+    accountAuthenticator.accountManager = accountManager
+  }
+
+  @Test
+  fun `verify add account`() {
```

Comment:
I think these tests can have the same naming convention eg `methodxyz should do when given`. This applies for all test methods. I think `verify` is misleading since we are not verifying calls were made to the method